js/ui/search.js

```diff
@@ -174,7 +174,7 @@ const ListSearchResult = new Lang.Class({
                                y_align: St.Align.MIDDLE });
 
         let title = new St.Label({ style_class: 'list-search-result-title',
-                                   text: this.metaInfo['name'],
+                                   text: this.metaInfo['name'] || '',
```

Comment:
I don't think anything major should happen, since I'm quite sure all result construction is within a try/catch. So you'll see errors, but they'll be properly caught. I think more than anything all the results for that app would be dead, which is a bit heavyhanded of a punishment. The logic is "if the app gave us any junk at all, it can't be trusted properly".

I don't like this option, because the user will just see an empty bubble as a result.